numpy/lib/polynomial.py

```diff
@@ -139,12 +139,11 @@ def poly(seq_of_zeros):
     if issubclass(a.dtype.type, NX.complexfloating):
         # if complex roots are all complex conjugates, the roots are real.
         roots = NX.asarray(seq_of_zeros, complex)
-        pos_roots = sort_complex(NX.compress(roots.imag > 0, roots))
-        neg_roots = NX.conjugate(sort_complex(
-                                        NX.compress(roots.imag < 0, roots)))
-        if (len(pos_roots) == len(neg_roots) and
-            NX.alltrue(neg_roots == pos_roots)):
-            a = a.real.copy()
+        pos_roots = NX.compress(roots.imag > 0, roots)
```

Comment:
I'd guess that going much over 20 is going to be a problem in general. I've gone up to 60 equally spaced roots in the polynomial package, which does better than this, but the results can be sensitive to root placement.